File: Classwork/AP_Computer_Science_Ⅰ/src/objectOriented/gridWorld/projects/myProjects/critters/caseStudyWork/chameleonCritters/fadingChameleonCritter/FadingChameleonCritter.java
package objectOriented.gridWorld.projects.myProjects.critters.caseStudyWork.chameleonCritters.fadingChameleonCritter;

import objectOriented.gridWorld.framework.info.gridworld.actor.Actor;
import objectOriented.gridWorld.projects.critters.chameleonCritter.ChameleonCritter;
import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.Random;

public class FadingChameleonCritter extends ChameleonCritter {
	@Override
	public void processActors(@NotNull ArrayList<Actor> actors) {
		setColor((actors.size() == 0) ? getColor().darker()
				: actors.get(new Random().nextInt(actors.size())).getColor());
	}
}

File: Classwork/AP_Computer_Science_Ⅰ/src/objectOriented/gridWorld/projects/myProjects/critters/caseStudyWork/chameleonCritters/fadingChameleonCritter/FadingChameleonCritter.kt
package objectOriented.gridWorld.projects.myProjects.critters.caseStudyWork.chameleonCritters.fadingChameleonCritter

import info.gridworld.actor.Actor
import objectOriented.gridWorld.projects.critters.chameleonCritter.ChameleonCritter
import java.util.*

open class FadingChameleonCritter : ChameleonCritter() {
	override fun processActors(actors: ArrayList<Actor>) {
		color = if (actors.size == 0) color.darker() else actors[Random().nextInt(actors.size)].color
	}
}
